code/game/objects/structures/dropship_ammo.dm

```diff
@@ -211,57 +210,33 @@
 	T.ignite(5, 30) //short but intense
 
 
-//Railgun
+//railgun
 
-/obj/structure/ship_ammo/railgun
+/obj/structure/ship_ammo/energy/railgun
 	name = "railgun ammunition unit"
 	icon_state = "laser_battery"
 	desc = "Changeable capacitors and projectiles clip in single unit, complexity of the design does not allow transfer of ammo. Used to fire railgun."
 	travelling_time = 2 SECONDS
-	ammo_count = 5
-	max_ammo_count = 5
 	equipment_type = /obj/structure/dropship_equipment/weapon/railgun
-	ammo_name = "projectile charges"
 	transferable_ammo = FALSE
-	ammo_used_per_firing = 1
-	warning_sound = 'sound/effects/nightvision.ogg'
+	ammo_used_per_firing = 20 // 5 shots
 	point_cost = 200
-	var/rail_radius = 1
+	impact_radius = 0
 	ammo_type = CAS_RAILGUN
 
-/obj/structure/ship_ammo/railgun/examine(mob/user)
+/obj/structure/ship_ammo/energy/railgun/examine(mob/user)
 	. = ..()
-	to_chat(user, "It's at [round(100*ammo_count/max_ammo_count)]% charge and [ammo_count] projectiles.")
+	to_chat(user, "It's at [round(100*ammo_count/max_ammo_count)]% charge and [ammo_count/20] projectiles.")
 
 
-/obj/structure/ship_ammo/railgun/show_loaded_desc(mob/user)
+/obj/structure/ship_ammo/energy/railgun/show_loaded_desc(mob/user)
 	if(ammo_count)
 		to_chat(user, "It's loaded with \a [src] at [round(100*ammo_count/max_ammo_count)]% charge and [ammo_count] projectiles.")
 	else
 		to_chat(user, "[name] is empty.")
 
-
-/obj/structure/ship_ammo/railgun/detonate_on(turf/impact, attackdir = NORTH)
-	var/turf/beginning = impact
-	var/turf/end = impact
-	var/revdir = REVERSE_DIR(attackdir)
-	for(var/i=0 to rail_radius)
-		beginning = get_step(beginning, revdir)
-		end = get_step(end, attackdir)
-	var/list/turf/railtargets = getline(beginning, end)
-	process_rail(railtargets)
-	if(!ammo_count)
-		QDEL_IN(src, rail_radius+1) //deleted after last is fired and impacts the ground
-
-///takes the top railtarget on the stack and fires the rail at it
-/obj/structure/ship_ammo/railgun/proc/process_rail(list/railtargets)
-	rail_hit(railtargets[1])
-	railtargets -= railtargets[1]
-	if(length(railtargets))
-		INVOKE_NEXT_TICK(src, .proc/process_rail, railtargets)
-
 ///Rail ammo acts on the turf passed in
-/obj/structure/ship_ammo/railgun/proc/rail_hit(turf/T)
+/obj/structure/ship_ammo/energy/railgun/projectile_impact(turf/T)
 	playsound(T, 'sound/effects/explosion6.ogg', 4, 1)
 	T.ex_act(EXPLODE_DEVASTATE)
```

Comment:
As said in to-do, it's too powerful and `EXPLODE_DEVASTATE` is too weak. Need help help here. I can find out how to define new and define it, adjusting as needed, but i'm not sure it's best way to do it.

Reply:
Currently experimenting with projectile system. May be it will be literally projectile, yes. `bullet_act()` looks versatile enough for xeno damage configurating. Other obj's, turf's working by `for (***/*** in T) ***.ex_act(EXPLODE_DEVASTATE)`.